prow/config/config.go

```diff
@@ -212,68 +213,72 @@ type InRepoConfig struct {
-	normalizedIdentifier := trimRepoPrefix(identifier)
-
-	if c.InRepoConfig.Enabled[normalizedIdentifier] != nil {
-		return *c.InRepoConfig.Enabled[normalizedIdentifier]
-	}
-
-	// Errors if failed to split. We are ignoring this and just checking if org != "" instead.
-	if org, _, _ := SplitRepoName(identifier); org != "" && c.InRepoConfig.Enabled[org] != nil {
-		return *c.InRepoConfig.Enabled[org]
-	}
-	if c.InRepoConfig.Enabled["*"] != nil {
-		return *c.InRepoConfig.Enabled["*"]
+	for _, key := range keysForIdentifier(identifier) {
+		if c.InRepoConfig.Enabled[key] != nil {
+			return *c.InRepoConfig.Enabled[key]
+		}
 	}
 	return false
 }
 
 // InRepoConfigAllowsCluster determines if a given cluster may be used for a given repository
 // Assumes that config will not include http:// or https://
-func (c *Config) InRepoConfigAllowsCluster(clusterName, repoIdentifier string) bool {
-	normalizedIdentifier := trimRepoPrefix(repoIdentifier)
-	for _, allowedCluster := range c.InRepoConfig.AllowedClusters[normalizedIdentifier] {
-		if allowedCluster == clusterName {
-			return true
-		}
-	}
-
-	// Errors if failed to split. We are ignoring this and just checking if org != "" instead.
-	if org, _, _ := SplitRepoName(repoIdentifier); org != "" {
-		for _, allowedCluster := range c.InRepoConfig.AllowedClusters[org] {
+func (c *Config) InRepoConfigAllowsCluster(clusterName, identifier string) bool {
+	for _, key := range keysForIdentifier(identifier) {
+		for _, allowedCluster := range c.InRepoConfig.AllowedClusters[key] {
 			if allowedCluster == clusterName {
 				return true
 			}
 		}
 	}
+	return false
+}
 
-	for _, allowedCluster := range c.InRepoConfig.AllowedClusters["*"] {
-		if allowedCluster == clusterName {
-			return true
-		}
+// keysForIdentifier returns all possible identifiers for given keys. In
+// consideration of Gerrit identifiers that contain `https://` prefix, it
+// returns keys contain both `https://foo/bar` and `foo/bar` for identifier
+// `https://foo/bar`. The returned keys also include `https://foo`, `foo`, and
+// `*`.
+func keysForIdentifier(identifier string) []string {
+	keySets := sets.NewString()
+
+	normalizedIdentifier := identifier
+	if gerritsource.IsGerritOrg(identifier) {
+		normalizedIdentifier = gerritsource.NormalizeCloneURI(identifier)
 	}
-	return false
+
+	keySets.Insert(normalizedIdentifier)
+	// gerritsource.TrimHTTPSPrefix(identifier) trims https:// prefix, it
+	// doesn't hurt for identifier without https://
+	keySets.Insert(gerritsource.TrimHTTPSPrefix(identifier))
+
+	org, _, _ := SplitRepoName(normalizedIdentifier)
+	// Errors if failed to split. We are ignoring this and just checking if org != "" instead.
+	if org != "" {
+		keySets.Insert(org)
+		// gerritsource.TrimHTTPSPrefix(identifier) trims https:// prefix, it
+		// doesn't hurt for identifier without https://
+		keySets.Insert(gerritsource.TrimHTTPSPrefix(org))
+	}
+
+	keySets.Insert("*")
+	return keySets.List()
```

Comment:
We can't use a set to collect the keys because the ordering of the list that is returned matters. Specifically we need to return the keys in order of most specific to most general other wise we might use the config for `*` when there is a more specific config for the org or repo.
Please add a unit test for InRepoConfigEnabled to catch this.

Reply:
done